lightstep/instrument.py

```diff
@@ -177,12 +178,27 @@ def _flush_worker(self, connection):
 
     def _construct_report_request(self):
         """Construct a report request."""
+        report = None
         with self._mutex:
             report = ttypes.ReportRequest(self._runtime, self._span_records,
                                           self._log_records)
             self._span_records = []
             self._log_records = []
-            return report
+
+        for log in report.log_records:
+            if log.stable_name is not None:
+                log.stable_name = str(log.stable_name)
+            if log.payload_json is not None:
+                try:
+                    log.payload_json = \
+                                       jsonpickle.encode(log.payload_json,
+                                                         unpicklable=False,
+                                                         make_refs=False,
+                                                         max_depth=constants.JSON_MAX_DEPTH)
+                except:
+                    log.payload_json = jsonpickle.encode(constants.JSON_FAIL)
```

Comment:
I think as a rule, we are trying to avoid putting internal errors in the `payload_json` field. I recommend chatting quickly with @will-lightstep as I think he had to do something here recently as well.